Mathlib/NumberTheory/Cyclotomic/Rat.lean

```diff
@@ -238,13 +238,14 @@ theorem power_basis_int'_dim [hcycl : IsCyclotomicExtension {p} ℚ K] (hζ : Is
     pow_one]
 #align is_primitive_root.power_basis_int'_dim IsPrimitiveRoot.power_basis_int'_dim
 
+instance : NoZeroSMulDivisors ℤ (𝓞 K) :=
+  inferInstanceAs (NoZeroSMulDivisors ℤ (integralClosure _ _))
```

Comment:
Shouldn't this be a consequence of [`NoZeroSMulDivisors.CharZero.noZeroSMulDivisors_int`](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Algebra/Basic.html#NoZeroSMulDivisors.CharZero.noZeroSMulDivisors_int)?

Reply:
I see, here we have `CharZero K` but the `CharZero (𝓞 K)` instance assumes `NumberField K`.